### PR TITLE
Add Channels.py classes for each Channel type (only __init__())

### PR DESCRIPTION
**Pull request by MartinHeroux**
Before going too far in coding and design, I wanted to push the template for the Channel classes. 

I have only included the special method __init__() in each class for now, and assigned some attributes. 

If this seems sensible, I will add Methods to some of the classes (most will go into the Waveform class).

### CODE REVIEW

**Comment by bbelderbos** [comment]
Technically this code looks correct, although as per training "interfaces should be small" so I am a bit worried about this class getting so many parameters:

```
    def __init__(self, units, template_length, times, sampling_frequency,
                  action_potentials, name, trial_name):
```

But I can only really judge this design if you can explain to me in layman's terms why you chose this structure? 

Do we really need 4 levels of inheritance here? 
Are all these classes dependent on each other, that is do all children need functionality of the parent / base classes? 
Or would it make sense to have the subclasses receive the (what are now parent) classes as mixins or init args?

**Comment by MartinHeroux** [comment]
I agree with your comment; I was actually reading the 'Building Maintainable Software' chapter on interfaces today!

Unfortunately I am not great with classes, so I am not totally sure what you mean by 4 levels of inheritance.

----
Channels -> Base class
                |---> Event -> inherits Channels
                           |---> All other classes -> inherit Event
----
The Channels base class is a bit silly, I agree. Everything could start at the Event class.

In terms of functionality:

- The Event class and the Keyboard class will likely have no methods.
- The Waveform class will have several methods, all related to signal processing.
- The Wavemark class will have very few methods; possibly just a special plotting method.

I am less clear what you mean by 'mixins' or 'init args'; I will try to read up on those. 

Note: The various arguments that are passed to the classes are actually the keys to the various dictionaries that are returned from the read.read.py. So I don't have to create an instance of the channel classes as follows:

```python 
data = read.read('somefile.mat')
biceps = Waveform(**data['biceps'])
```
I could do it this way:
```python 
data = read.read('somefile.mat')
biceps = Waveform(data['biceps'])
```
Then, in the __init__ of the class I could parse the input:

```python
class Waveform(Event):
def __init__(self, waveform_dict):
    self.name = waveform_dict['name']
    self.times = waveform_dict['times']
    self.values = waveform_dict['values']
    # etc
```

Not sure this will make sense to you...

**Comment by bbelderbos** [comment]
We can discuss in a few hours, I think we need to clarify which classes barrow functionality from which other classes, if they do, it makes sense to use inheritance.

**Comment by bbelderbos** [comment]
This seems a bit verbose / redundant to me:

```
    self.name = waveform_dict['name']
    self.times = waveform_dict['times']
    self.values = waveform_dict['values']
```

Let's discuss later how we can best do this ...

And cool you are reading that book, learning a lot from it?
